src/libraries/System.Private.CoreLib/src/System/Buffers/Binary/WriterBigEndian.cs

```diff
@@ -31,6 +31,29 @@ public static void WriteDoubleBigEndian(Span<byte> destination, double value)
             }
         }
 
+        /// <summary>
+        /// Writes a <see cref="Half" /> into a span of bytes, as big endian.
+        /// </summary>
+        /// <param name="destination">The span of bytes where the value is to be written, as big endian.</param>
+        /// <param name="value">The value to write into the span of bytes.</param>
+        /// <remarks>Writes exactly 2 bytes to the beginning of the span.</remarks>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="destination" /> is too small to contain a <see cref="Half" />.
+        /// </exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void WriteHalfBigEndian(Span<byte> destination, Half value)
+        {
+            if (BitConverter.IsLittleEndian)
+            {
+                short tmp = ReverseEndianness(BitConverter.HalfToInt16Bits(value));
+                MemoryMarshal.Write(destination, ref tmp);
+            }
+            else
+            {
+                MemoryMarshal.Write(destination, ref value);
+            }
```

Comment:
very-very pedantic nit: you are sometimes using `else` block, sometimes not`

```cs
if (condition)
{
   // do one thing
   // return sth;
}

// return sth else
```

```cs
if (condition)
{
   // do one thing
   // return sth;
}
else
{
   // return sth else
}
```

it would be good to stick to one pattern and follow it in the entire PR